### PR TITLE
Query collection and collection activities

### PR DESCRIPTION
**Pull request by alchemistgo87**

### CODE REVIEW

File: crates/graphql/src/schema/objects/nft.rs [hunk 1]
```diff
@@ -537,6 +537,22 @@ impl Collection {
...
 
+    pub async fn activities(
+        &self,
+        ctx: &AppContext,
+        limit: i32,
+        offset: i32,
+    ) -> FieldResult<Vec<NftActivity>> {
+        let conn = ctx.shared.db.get()?;
+        let rows =
+            queries::metadatas::collection_activities(&conn, &self.0.address, limit, offset)?;
```
Comment (kespinola) [inline crates/graphql/src/schema/objects/nft.rs:548]: This is going to be an n+1 query but we don't currently have it speced in the UI to display multiple collection activities at the same time so advocate to keep and address if the requirements change.

Comment (kespinola) [reply crates/graphql/src/schema/objects/nft.rs:548]: If you can do with a data loader please give it a try.

File: crates/graphql/src/schema/objects/nft.rs [hunk 2]
```diff
@@ -655,13 +671,13 @@ impl Collection {
...
-            .await
-            .map_err(Into::into)
-    }
+    // #[graphql(deprecated = "use `nft { activities }`")]
+    // pub async fn activities(&self, ctx: &AppContext) -> FieldResult<Vec<NftActivity>> {
+    //     ctx.nft_activities_loader
+    //         .load(self.0.address.clone().into())
+    //         .await
+    //         .map_err(Into::into)
+    // }
```
Comment (kespinola) [inline crates/graphql/src/schema/objects/nft.rs:680]: delete lines.

File: crates/graphql/src/schema/query_root.rs [hunk 3]
```diff
@@ -720,6 +721,40 @@ impl QueryRoot {
...
+            .filter(metadata_collection_keys::collection_address.eq(address))
+            .filter(metadata_collection_keys::verified.eq(true))
+            .select(queries::metadatas::NFT_COLUMNS)
+            .first::<models::Nft>(&conn)
+            .optional()
+            .context("Failed to load NFT by metadata address.")?
+            .map(TryInto::try_into)
+            .transpose()
+            .map_err(Into::into)
+    }
```
Comment (kespinola) [inline crates/graphql/src/schema/query_root.rs:756]: Can you please bundle this query in the queries::collection module under show function? Keep the query root more readable that way.

File: crates/core/src/db/queries/metadatas.rs [hunk 4]
```diff
@@ -468,3 +468,55 @@ pub fn activities(
...
+    limit: impl ToSql<Integer, Pg>,
+    offset: impl ToSql<Integer, Pg>,
+) -> Result<Vec<NftActivity>> {
+    diesel::sql_query(COLLECTION_ACTIVITES_QUERY)
+        .bind(address)
+        .bind(limit)
+        .bind(offset)
+        .load(conn)
+        .context("Failed to load collection activities")
+}
```
Comment (kespinola) [inline crates/core/src/db/queries/metadatas.rs:522]: Lets move to queries::collection::activites()

The UI also requires some filtering where we need to support include_types: ["LISTINGS", "PURCHASES", "OFFERS"]

**Review by kespinola — CHANGES_REQUESTED** [review]
Looks good some items to clean up and missing requirement on the included_types.

File: crates/graphql/src/schema/objects/nft.rs (hunk 1 above)
Comment (alchemistgo87) [reply crates/graphql/src/schema/objects/nft.rs:548]: If I create a collection_activities loader, we will have 2 loaders of same type:

```    
pub collection_activities_loader: Loader<PublicKey<Nft>, Vec<NftActivity>>,

pub nft_activities_loader: Loader<PublicKey<Nft>, Vec<NftActivity>>,
```

**Review by kespinola — APPROVED** [review]

**Review by kristianeboe — APPROVED** [review]

**Pull request merged**